sources/packages/web/src/components/common/IdleTimeChecker.vue

```diff
@@ -57,44 +84,114 @@ export default {
-        /* eslint-enable */
+        interval.value = setInterval(checkIdle, 1000);
       }
     };
 
+    watch(
+      () => modalOpen.value,
+      (newValue: boolean) => {
+        if (newValue) {
+          countdownInterval.value = setInterval(() => {
+            if (countdown.value > 0) {
+              countdown.value = maximumIdleTime.value - idleTimeInSeconds.value;
+            }
+          }, 1000);
+        } else {
+          clearInterval(countdownInterval.value);
+          countdown.value = COUNT_DOWN_TIMER_FOR_LOGOUT;
+        }
+      },
+    );
+
+    const extendUserSessionTime = () => {
+      setLastActivityTime();
+      resetIdleCheckerTimer();
+      modalOpen.value = false;
+      clearInterval(countdownInterval.value);
+      countdown.value = COUNT_DOWN_TIMER_FOR_LOGOUT;
+    };
+
     const confirmExtendTimeModal = async () => {
+      modalOpen.value = true;
       if (await extendTimeModal.value.showModal()) {
-        lastActivityLogin.value = new Date();
-        resetIdleCheckerTimer();
+        extendUserSessionTime();
+      } else {
+        setLoggedOut();
       }
     };
 
     const checkIdle = () => {
-      const idleTimeInMintutes = getDatesDiff(
-        lastActivityLogin.value,
+      if (getLoggedOut()) {
+        logoff();
+      }
+      idleTimeInSeconds.value = getDatesDiff(
+        getLastActivityTime(),
         new Date(),
         "second",
-        true,
+        false,
       );
-      if (idleTimeInMintutes >= minimumIdleTime.value) {
+
+      // Exceeded user session time.
+      if (idleTimeInSeconds.value > maximumIdleTime.value) {
+        if (modalOpen.value === true) {
+          modalOpen.value = false;
+          clearInterval(countdownInterval.value);
+        }
+        // Logoff immediately in case session is expired.
+        logoff();
+      } else if (
+        idleTimeInSeconds.value >=
+        maximumIdleTime.value - COUNT_DOWN_TIMER_FOR_LOGOUT
+      ) {
         confirmExtendTimeModal();
+      } else {
+        // Close the modal in case idleTimeInSeconds has changed
+        // because lastAcitivityTime changed from activity in other tab/window.
+        modalOpen.value = false;
       }
     };
 
-    onMounted(async () => {
-      resetIdleCheckerTimer();
-    });
-
     const setLastActivityTime = () => {
       if (isAuthenticated.value) {
-        lastActivityLogin.value = new Date();
+        localStorage.setItem(
+          LAST_ACTIVITY_TIME_LOCAL_STORAGE_ITEM,
+          new Date().getTime().toString(),
+        );
       }
     };
+
+    const getLastActivityTime = () => {
+      const localStorageValue = localStorage.getItem(
+        LAST_ACTIVITY_TIME_LOCAL_STORAGE_ITEM,
+      );
+      if (localStorageValue) {
+        return new Date(+localStorageValue);
+      }
+      return new Date();
+    };
+
+    const setLoggedOut = () => {
+      localStorage.setItem(LOGGED_OUT_LOCAL_STORAGE_ITEM, "true");
```

Comment:
If the user is logged off and login again, does it need to be reset?

Reply:
I got the answer to this. It is reset during the `onMount` and since every time that we do a new login, we leave the portal and reload the entire portal the component will also be reloaded.